### DISCUSSION TITLE
Работа youtube на ТВ Samsung (Tizen)

### DISCUSSION THREAD

**Discussion by sintezenish** [bol-van/zapret#851] [Q&A] [answered]
Сейчас конфиг выглядит так

NFQWS_OPT_DESYNC="--dpi-desync=fake,split --dpi-desync-repeats=6 --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --dpi-desync-fooling=badseq"

NFQWS_OPT_DESYNC_QUIC="--dpi-desync=fake --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --dpi-desync-repeats=6"

На win и android все работает без замечаний. Сложности возникают на телевизоре. Там пишет при загрузке youtube - Ошибка сети, и приложение вываливается.

Если использовать встроенный браузер - просто не может зайти по адресу.

Посоветуйте, что можно попробовать ?

**Comment by hlivergaing** [comment]
Аналогичная проблема , неделю назад работало везде. А сейчас на WEB OS LG TV перестало работать.

**Comment by andmt1** [comment]
Писали уже тут - увеличьте параметр --dpi-desync-repeats до 15. дб: --dpi-desync-repeats=15
и split просто не работает. попробуйте так: --dpi-desync=fake,disorder2
а на quic (на нем и работает ТВ): --dpi-desync=fake,split2 --dpi-desync-repeats=15

**Comment by sintezenish** [comment]
Попробовал. maps.google.com не открывается на пк
На ТВ вылетает само приложение с ошибкой сети
Стоят такие параметры

NFQWS_OPT_DESYNC="--dpi-desync=fake,disorder2 --dpi-desync-repeats=15 --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --dpi-desync-fooling=badseq"


NFQWS_OPT_DESYNC_QUIC="--dpi-desync=fake,split2 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --dpi-desync-repeats=15"

**Comment by 9iky6** [comment]
Попробуйте этот вариант.
NFQWS_OPT_DESYNC="--dpi-desync=fake,disorder2 --dpi-desync-repeats=20 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin"

**Comment by andmt1** [comment]
На РТ так работает:
NFQWS_OPT_DESYNC="--dpi-desync=fake,disorder2 --dpi-desync-repeats=15 --dpi-desync-ttl=0 --dpi-desync-fooling=md5sig,badsum --dpi-desync-repeats=15 --dpi-desync-any-protocol --dpi-desync-cutoff=d4 --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin"
NFQWS_OPT_DESYNC_QUIC="--dpi-desync=fake,split2 --dpi-desync-repeats=15 --dpi-desync-ttl=0 --dpi-desync-any-protocol --dpi-desync-cutoff=d4 --dpi-desync-fooling=md5sig,badsum --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_google_com.bin"

**Comment by sintezenish** [comment]
Не работает. Приложение сразу вываливается с ошибкой сети

**Comment by andmt1** [comment]
на ТВ остальные приложения работают?

**Comment by sintezenish** [comment]
Да. Я игрался с параметрами. Сейчас стоит так:
NFQWS_OPT_DESYNC="--dpi-desync=fake,split2 --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --dpi-desync-fooling=badsum --dpi-desync-repeats=20" 

NFQWS_OPT_DESYNC_QUIC="--dpi-desync=fake,split2 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --dpi-desync-repeats=20"


Приложение открывается, привью есть, но видео не стартует.
если опускать --dpi-desync-repeats=20 ниже 15, приложение на ТВ вылетает

возможно еще какой то параметр влияет...

**Comment by andmt1** [comment]
на приложение  влияет эта строка - NFQWS_OPT_DESYNC
на загрузку видео - эта - NFQWS_OPT_DESYNC_QUIC
надо бы вам попробовать блокчеком проверить googlevideo.com

**Comment by bol-van** [comment]
Прямая проверка googlevideo.com бесполезна, она ничего не даст, будет ошибка на каждую попытку и 0 в результате.
Надо проверять GGC, адреса которых можно получить по F12 в броузере или по результатам перехвата в tcpdump на роутере

**Comment by sintezenish** [comment]
Где об этом можно почитать, как правильно нужно сделать что бы получить результат ?
Вы правы, простая проверка параметров не выдала.

**Comment by slavroman** [comment]
> Где об этом можно почитать, как правильно нужно сделать что бы получить результат ? Вы правы, простая проверка параметров не выдала.

https://ntc.party/t/%D0%BF%D0%BE%D0%B4%D0%B1%D0%BE%D1%80-%D1%80%D0%B0%D0%B1%D0%BE%D1%87%D0%B5%D0%B3%D0%BE-%D0%BA%D0%BE%D0%BD%D1%84%D0%B8%D0%B3%D0%B0-%D0%B4%D0%BB%D1%8F-ggc-%D1%8E%D1%82%D1%83%D0%B1%D0%B0-%D1%87%D0%B5%D1%80%D0%B5%D0%B7-blockcheck/10060

https://ntc.party/t/goodcheck-%D0%B1%D0%BB%D0%BE%D0%BA%D1%87%D0%B5%D0%BA-%D1%81%D0%BA%D1%80%D0%B8%D0%BF%D1%82-%D0%B4%D0%BB%D1%8F-goodbyedpi-zapret-byedpi/10880

P.S.>Вроде бы интернет у всех одинаковый, а какие разные навыки поиска...

**Comment by sintezenish** [comment]
Почитал. Запустил. 

Получился конфиг. Строчку взял из результата запуска анализа 
NFQWS_OPT_DESYNC="--dpi-desync=fake,split --dpi-desync-ttl=3 --dpi-desync-fake-tls=0x00000000 --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin"

NFQWS_OPT_DESYNC_QUIC="--dpi-desync=fake,split2 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --dpi-desync-repeats=15"

Это оставил из предыдущего конфига

На ПК/телефоне - работает. На ТВ сайт недоступен, приложение выкидывает с ошибкой

**Comment by DRGremlinz** [comment]
так есть способ решения? везде все работает а на tizen нет.

**Comment by andmt1** [comment]
Настривать DoH или DoQ. Ставить dnsproxy и Dnsmasq и блокировать 53 порт. (https-dns-proxy не надо, он не работает с quic)
https://openwrt.org/docs/guide-user/services/dns/dot_dnsmasq_dnsproxy

Doq сервер Cloudflare - h3://security.сloudflare-dns.com/dns-query 
он H3 но работает по протоколу Quic
или quic://dns.adguard-dns.com

**Comment by numbereleven-a** [comment]
> Настривать DoH или DoQ. Ставить dnsproxy и Dnsmasq и блокировать 53 порт. (https-dns-proxy не надо, он не работает с quic) https://openwrt.org/docs/guide-user/services/dns/dot_dnsmasq_dnsproxy
> 
> Doq сервер Cloudflare - h3://security.сloudflare-dns.com/dns-query он H3 но работает по протоколу Quic или quic://dns.adguard-dns.com

а почему решили что на тизенском ютубе используется h3?
у меня квик включен и дурится, но труба на тизене использует tcp

IP4: 192.168.1.8 => 172.217.133.74 proto=tcp ttl=63 sport=55259 dport=443 flags=S
desync profile search for tcp target=172.217.133.74:443 l7proto=tls hostname='r5---sn-f5f7lne6.c.youtube.com'

IP4: 192.168.1.8 => 212.188.37.17 proto=tcp ttl=63 sport=47861 dport=443 flags=S
desync profile search for tcp target=212.188.37.17:443 l7proto=tls hostname='rr6---sn-jvhnu5g-c35d.googlevideo.com'

Или это надо как то по другому проверять?

**Comment by andmt1** [comment]
речь про шифрование DNS (quic здесь между вашим роутером и днс сервером). Какой протокол использует тизен я не в курсе, предполагаю, что оболочка по tcp, а видео через quic. Более точно можно посмотреть через wireshark

**Comment by numbereleven-a** [comment]
> речь про шифрование DNS (quic здесь между вашим роутером и днс сервером). Какой протокол использует тизен я не в курсе, предполагаю, что оболочка по tcp, а видео через quic. Более точно можно посмотреть через wireshark

rr6---sn-jvhnu5g-c35d.googlevideo.com - вот же сервер видео и он по tcp, если у людей не грузится сам ютуб а не видео, там тоже tcp
причём тут dns хз, ну ладно, могу лишь сказать что на тизене в моём случае каких то особенных стратегий и протоколов выбирать не нужно

**Comment by andmt1** [comment]
> причём тут dns хз,

а вы проверьте...

провайдер замедляет googlevideo.com анализируя dns запросы или подменяя прописанный у вас dns сервер на свой. (это помимо анализа sni)
Без шифрования DNS zapret полноценно не будет работать на любых настройках

**Comment by numbereleven-a** [comment]
> а вы проверьте...

я имею ввиду причём тут dns если на тизене не работает, а в телефоне и на компе работает.

**Comment by sintezenish** [comment]
> > причём тут dns хз,
> 
> а вы проверьте...
> 
> провайдер замедляет googlevideo.com анализируя dns запросы или подменяя прописанный у вас dns сервер на свой. (это помимо анализа sni) Без шифрования DNS zapret полноценно не будет работать на любых настройках

Например я получаю результат
* SUMMARY
ipv4 rr2---sn-jvhnu5g-c35d.googlevideo.com curl_test_http : working without bypass
ipv4 rr2---sn-jvhnu5g-c35d.googlevideo.com curl_test_https_tls12 : tpws --split-tls=sni --disorder --mss=88
ipv4 rr2---sn-jvhnu5g-c35d.googlevideo.com curl_test_https_tls12 : nfqws --dpi-desync=split --dpi-desync-ttl=4
ipv4 rr2---sn-jvhnu5g-c35d.googlevideo.com curl_test_https_tls13 : tpws --tlsrec=sni --split-pos=1 --oob
ipv4 rr2---sn-jvhnu5g-c35d.googlevideo.com curl_test_https_tls13 : nfqws --dpi-desync=split --dpi-desync-fooling=badseq

что с ним делать ?

**Comment by andmt1** [comment]
домен googlevideo.com не заблокирован и тест не даст результата

**Comment by sintezenish** [comment]
Нужно смотреть какой у вас хост.

**Comment by sintezenish** [comment]
Долгое время на 68 версии работало так

NFQWS_OPT="
--filter-udp=443 --hostlist=/opt/zapret-lists/list-global.txt --dpi-desync=fake --dpi-desync-repeats=9 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --new
--filter-tcp=443 --hostlist=/opt/zapret-lists/list-global.txt --dpi-desync=fake,split2 --dpi-desync-split-seqovl=1 --dpi-desync-split-tls=sniext --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --dpi-desync-ttl=4 --new
--filter-tcp=443 --hostlist=/opt/zapret-lists/list-youtube.txt --dpi-desync=fake,split2 --dpi-desync-split-seqovl=1 --dpi-desync-repeats=11 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --new
--filter-udp=443 --hostlist=/opt/zapret-lists/list-discord.txt --dpi-desync=fake,disorder2 --dpi-desync-cutoff=d4 --dpi-desync-udplen-increment=10 --dpi-desync-repeats=6 --dpi-desync-udplen-pattern=0xDEADBEEF --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --new
--filter-tcp=443 --hostlist=/opt/zapret-lists/list-discord.txt --dpi-desync=fake,disorder2 --dpi-desync-autottl=1 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --new
--filter-udp=50000-65535 --dpi-desync=fake,tamper --dpi-desync-any-protocol --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin
"

После обновления на последнюю версию 71.1.1 эта стратегия перестала работать
Конфиг
NFQWS_OPT="
--filter-tcp=80 --dpi-desync=fake,multisplit --dpi-desync-ttl=0 --dpi-desync-fooling=md5sig,badsum --dpi-desync-fake-http=/opt/zapret/files/fake/dht_get_peers.bin --new
--filter-tcp=443 --dpi-desync=fake,multidisorder --dpi-desync-split-pos=method+2,midsld,5 --dpi-desync-ttl=0 --dpi-desync-fooling=md5sig,badsum,badseq --dpi-desync-repeats=15 --dpi-desync-any-protocol --dpi-desync-cutoff=d4 --dpi-desync-fake-tls=/opt/zapret/files/fake/dht_get_peers.bin --new
--filter-udp=443 --dpi-desync=fake --dpi-desync-repeats=15 --dpi-desync-ttl=0 --dpi-desync-any-protocol --dpi-desync-cutoff=d4 --dpi-desync-fooling=md5sig,badsum --dpi-desync-fake-quic=/opt/zapret/files/fake/dht_get_peers.bin --new
--filter-udp=50000-50099 --filter-l7=discord,stun --dpi-desync=fake
"

Дает возможность просмотра на телефоне, компе. Кроме Samsung Smart TV на Tizen. Выдает Ошибка сети.

Кто сталкивался или может быть есть идеи как можно поправить конфиг, что бы заработало на ТВ ?

**Reply by sintezenish** [reply, marked as answer]
Отвечу сам на свой вопрос. Решил не дожидаться решений других, и самостоятельно покопать. Заодно использовать chatgtp. Я ему показал старый конфиг и новый. Он мне выдал



📌 Если телевизор не переваривает dpi-desync=fake,split2, попробуй заменить на просто fake.

**Reply by AlexandrPravdin** [reply]
В итоге заработало?)

Заметил, что ломается какой-то конкретный запрос при запуске... 
Заметка:
Если запустить телевизор с впн (тыкал через подкоп) и потом выключить впн, то видео работают и все хорошо))

**Reply by Forenok** [reply]
> В итоге заработало?)
> 
> Заметил, что ломается какой-то конкретный запрос при запуске... Заметка: Если запустить телевизор с впн (тыкал через подкоп) и потом выключить впн, то видео работают и все хорошо))

До этого черный экран был при запуске ютуба.
Ставил wssize, который лучше на постоянку не использовать, с ним приложение грузилось, убирал wssize, около суток(вероятно 12 часов, на которые роутер присваивал ip телевизору) ютуб работал, потом опять черный экран при инициализации приложения. Поэтому да, при запуске приложения оно делает какой-то запрос, на котором с "тяжёлой" для телевизора стратегией - сеть отваливается.
У меня с этой стратегией(под спойлером) - заработало.
<details><summary>Стратегия</summary>

--filter-tcp=443
--hostlist=/opt/zapret/ipset/zapret-hosts-google.txt
--dpi-desync=fake,multisplit
--dpi-desync-fake-tls=0x0F0F0F0F
--dpi-desync-fooling=badseq
--dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin

--new
--filter-tcp=443 <HOSTLIST>
--dpi-desync=hostfakesplit
--dpi-desync-hostfakesplit-mod=host=rzd.ru
--dpi-desync-hostfakesplit-midhost=host-2
--dpi-desync-split-seqovl=726
--dpi-desync-fooling=badsum,badseq
--dpi-desync-badseq-increment=0

--new
--filter-udp=443 <HOSTLIST_NOAUTO>
--dpi-desync=fake,multidisorder
--dpi-desync-fooling=md5sig,badseq
--dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin

--new
--filter-udp=19294-19344,50000-50100
--filter-l7=discord,stun
--dpi-desync=fake
--dpi-desync-repeats=6

--new
--filter-tcp=2053,2083,2087,2096,8443
--hostlist-domains=discord.media
--dpi-desync=multisplit
--dpi-desync-split-seqovl=652
--dpi-desync-split-pos=2
--dpi-desync-split-seqovl-pattern=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin
</details>
